.github/workflows/build.yml

```diff
@@ -78,7 +78,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ ubuntu-20.04, ubuntu-22.04, ubuntu-24.04, macos-12, macos-14 ]
+        os: [ ubuntu-20.04, ubuntu-22.04, ubuntu-24.04, macos-13, macos-14, macos-15, macos-latest ]
```

Comment:
We want fixed versions here, specifically the minimum versions available, considering each architecture (amd64 & arm64).
The names of the produced files must remain exactly the same, see https://github.com/ruby/ruby-builder?tab=readme-ov-file#naming

Reply:
Since I'll need to do this for all the other repos I can do this one too